### PR TITLE
feat(3mf-deserializer): initial version of 3MF deserializer and tests

### PR DESCRIPTION
**Pull request by z3dev**
This is the new deserializer for 3MF packages or XML strings. This deserializer relies upon two packages to unzip 3MF packages (fflate) and parse XML (saxes). See the README for more information.

Testing has been completed with positive tests for 'core' functionality and 'materials' functionality. All 3MF files come from the 3MF test suite.

There's probably more 3MF features to support but this is a good starting point.

### All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [X] Does your submission pass tests?

### CODE REVIEW

**Review by hrgdavor — DISMISSED** [review]
I am not good at reviewwing. Looks good to me.

File: packages/io/3mf-deserializer/src/model.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      component.sequence = buildItems.length
+      buildItems.push(component)
+    }
+  })
+
+  if (includedItems === 'all') {
+    // FIXME loop through objects.. include anything missing
+  }
+
+  // console.log(model)
```
Comment (platypii) [inline packages/io/3mf-deserializer/src/model.js:37]: remove comments?

**Review by platypii — DISMISSED** [review]
Nice addition, code looks good.

File: packages/io/3mf-deserializer/src/model.js (hunk 1 above)
Comment (z3dev) [reply packages/io/3mf-deserializer/src/model.js]: @platypii thanks for the review.

Documentation has been improved, comments in code removed, and JSDOC configured.

**Comment by z3dev** [comment]
> I am not good at reviewwing. Looks good to me.

@hrgdavor You underestimate yourself. :)

all comments are welcome. and all improvements are welcome.

**Pull request merged**